src/bin/sol-fbp-generator/type-store.c

```diff
@@ -157,6 +159,30 @@ get_string(struct sol_json_token *token)
     return strndup(token->start + 1, token->end - token->start - 2);
 }
 
+static int
+get_int(struct sol_json_token *token, int *value)
+{
+    int v;
+    char *str, *endptr = NULL;
+
+    str = strndupa(token->start, token->end - token->start);
```

Comment:
There's no need to copy the string: call `strtol()`, and check if `endptr` points to `token->end` (if not, there was a parsing error).